custom_components/waste_collection_schedule/waste_collection_schedule/source/midsussex_gov_uk.py

```diff
@@ -45,49 +45,86 @@ def __init__(
     def fetch(self):
         s = requests.Session()
 
+        # Best practise scraper behaviour to specify own user-agent (allowing site owners to contact us if there is an issue)
+        s.headers.update({"User-Agent": "Home Assistant Waste Collection Schedule"})
+
         if self._address != "":
             # extract postcode
             self._postcode = re.findall(REGEX, self._address)
         elif self._house_name == "":
             self._address = (
-                self._house_number + " " + self._street + " " + self._postcode
+                self._house_number + ", " + self._street + ", " + self._postcode
+            )
+        elif self._house_number == "":
+            self._address = (
+                self._house_name + ", " + self._street + ", " + self._postcode
```

Comment:
This case needed to be handled specifically due to addition of the commas in the `else` below.